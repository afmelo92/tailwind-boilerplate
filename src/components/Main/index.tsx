import React from 'react'

export interface MainProps {
  title: string
}

const Main: React.FC<MainProps> = ({ title = 'Tailwind Boilerplate App' }) => {
  return (
    <main>
      <h1 className="text-9xl font-bold">{title}</h1>
    </main>
  )
}

export default Main
