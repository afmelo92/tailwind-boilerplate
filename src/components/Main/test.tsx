import { render, screen } from '@testing-library/react'

import Main from '.'

describe('<Main />', () => {
  it('should render heading', () => {
    const { container } = render(<Main />)

    expect(
      screen.getByRole('heading', { name: /tailwind boilerplate app/i })
    ).toBeInTheDocument()

    expect(screen.getByRole('heading', { name: /tailwind boilerplate app/i })).toHaveClass(
      'text-9xl font-bold'
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
