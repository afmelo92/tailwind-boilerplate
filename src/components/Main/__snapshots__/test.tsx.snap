// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render heading 1`] = `
<main>
  <h1
    class="text-9xl font-bold"
  >
    Tailwind Boilerplate App
  </h1>
</main>
`;
